src/spot.ts

```diff
@@ -0,0 +1,72 @@
+require("dotenv").load();
+import temp from "temp";
+import request from "request";
+
+interface IRequestOptsHeaders {
+  Authorization: string;
+}
+
+interface IRequestOpts {
+  headers: IRequestOptsHeaders;
+}
+
+let REQUEST_OPTS: IRequestOpts | null = null;
+let IMG_DIR: string | null = null;
+let SOUND_DIR: string | null = null;
+
+const setup = (): void => {
+  getToken();
+  temp.track();
+  temp.mkdir("images", (err, dirPath): void => {
+    IMG_DIR = dirPath;
+  });
+  temp.mkdir("sound", (err, dirPath): void => {
+    SOUND_DIR = dirPath;
+  });
+};
+
+interface IRequestForm {
+  grant_type: string;
+}
+
+// Request Lib may have a type for this
+interface IRequestOptions {
+  url: string;
+  headers: IRequestOptsHeaders;
+  json: boolean;
+}
+
+interface IAuthOptions extends IRequestOptions {
+  form: IRequestForm;
+}
+
+const getToken = (): void => {
+  const authOptions: IAuthOptions = {
+    url: "https://accounts.spotify.com/api/token",
```

Comment:
I reviewed the docs and this auth flow seems to be relatively unchanged. Have not investigated beyond general user grabbing ie grabbing playlists.